src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -412,6 +413,33 @@ private void validateShortName(Course course) {
         return userCourses.collect(Collectors.toList());
     }
 
+    /**
+     * GET /courses/courses-with-quiz : get all courses with quiz exercises for administration purposes.
+     *
+     * @return the list of courses
+     */
+    @GetMapping("/courses/courses-with-quiz")
+    @PreAuthorize("hasAnyRole('INSTRUCTOR', 'ADMIN')")
+    public List<Course> getAllCoursesWithQuizExercises() {
+        List<Course> courses;
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        if (authCheckService.isAdmin(user)) {
+            courses = courseRepository.findAll();
+        }
+        else {
+            courses = courseRepository.getCoursesForWhichUserHasInstructorAccess(user.getId());
+        }
+
+        List<Course> coursesWithQuiz = new ArrayList<>();
+        courses.forEach(course -> {
+            Course courseWithExercises = courseRepository.findWithEagerExercisesById(course.getId());
```

Comment:
this will not lead to a good performance, we should improve this in a follow up PR